core/src/main/kotlin/Utilities/DownloadSamples.kt

```diff
@@ -0,0 +1,64 @@
+package org.jetbrains.dokka.Utilities
+
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.File
+import java.io.FileOutputStream
+
+object DownloadSamples {
+
+    val client = OkHttpClient()
+
+    fun downloadSamples(): Boolean {
+
+        //loop through each directory of AOSP code in SamplesPathsToURLs.kt
+        filepathsToUrls.forEach { (filepath, url) ->
+
+            //build request using each URL
+            val request = Request.Builder()
+                .url(url)
+                .build()
+            val response = client.newCall(request).execute()
+
+            if (response.isSuccessful) {
+
+                //save .tar.gz file to filepath designated by map
+                var currentFile = File(filepath)
+                currentFile.mkdirs()
+
+                var fos = FileOutputStream(filepath + ".tar.gz")
```

Comment:
For Kotlin, you can use [string interpolation](https://kotlinlang.org/docs/reference/idioms.html#string-interpolation) over string concatenation